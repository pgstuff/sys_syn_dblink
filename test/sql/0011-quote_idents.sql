BEGIN;

CREATE EXTENSION sys_syn_dblink;

CREATE SCHEMA "Processor Schema"
        AUTHORIZATION postgres;

SELECT  dblink_connect('SysSyn Data', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)));
SELECT dblink_exec('SysSyn Data', 'BEGIN');

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('In Group');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('Out Group');

SELECT sys_syn_dblink.processing_table_add (
        schema          => '"Processor Schema"'::regnamespace,
        in_table_id     => 'Test Table',
        out_group_id    => 'Out Group',
        in_group_id     => 'In Group',
        dblink_connname => 'SysSyn Data');


SELECT * FROM "Processor Schema"."Test Table_Out Group_claim"();

SELECT * FROM "Processor Schema"."Test Table_Out Group_pull"();

SELECT * FROM "Processor Schema"."Test Table_Out Group_process"();

SELECT * FROM "Processor Schema"."Test Table_Out Group_push_status"();


SELECT dblink_exec('SysSyn Data', 'ROLLBACK');
SELECT dblink_disconnect('SysSyn Data');

ROLLBACK;
