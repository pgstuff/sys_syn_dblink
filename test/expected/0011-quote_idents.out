BEGIN;
CREATE EXTENSION sys_syn_dblink;
CREATE SCHEMA "Processor Schema"
        AUTHORIZATION postgres;
SELECT  dblink_connect('SysSyn Data', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)));
 dblink_connect 
----------------
 OK
(1 row)

SELECT dblink_exec('SysSyn Data', 'BEGIN');
 dblink_exec 
-------------
 BEGIN
(1 row)

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('SysSyn Data', 'In Group');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('SysSyn Data', 'Out Group');
INSERT INTO sys_syn_dblink.put_groups_def VALUES ('Put Group');
SELECT sys_syn_dblink.processing_table_add (
        proc_schema     => '"Processor Schema"'::regnamespace,
        in_table_id     => 'Test Table',
        out_group_id    => 'Out Group',
        put_group_id    => 'Put Group',
        dblink_connname => 'SysSyn Data');
 processing_table_add 
----------------------
 
(1 row)

SELECT * FROM "Processor Schema"."Test Table_Out Group_claim"();
 Test Table_Out Group_claim 
----------------------------
 t
(1 row)

SELECT * FROM "Processor Schema"."Test Table_Out Group_pull"();
 Test Table_Out Group_pull 
---------------------------
 t
(1 row)

SELECT * FROM "Processor Schema"."Test Table_Out Group_process"();
 Test Table_Out Group_process 
------------------------------
 t
(1 row)

SELECT * FROM "Processor Schema"."Test Table_Out Group_push_status"();
 Test Table_Out Group_push_status 
----------------------------------
 t
(1 row)

SELECT dblink_exec('SysSyn Data', 'ROLLBACK');
 dblink_exec 
-------------
 ROLLBACK
(1 row)

SELECT dblink_disconnect('SysSyn Data');
 dblink_disconnect 
-------------------
 OK
(1 row)

ROLLBACK;
